external-crates/move/crates/move-package/src/resolution/resolution_graph.rs

```diff
@@ -110,12 +112,28 @@ impl ResolvedGraph {
             let mut resolved_pkg = Package::new(package_path, &build_options)
                 .with_context(|| format!("Resolving package '{pkg_name}'"))?;
 
-            if pkg_name != resolved_pkg.source_package.package.name {
-                bail!(
-                    "Name of dependency '{}' does not match dependency's package name '{}'",
-                    pkg_name,
-                    resolved_pkg.source_package.package.name
-                )
+            // Check dependencies package names from manifest are consistent with ther names
+            // in parent (this) manifest. We do this check only for local and git
+            // dependencies as we assume custom dependencies might not have a user-defined
+            // name.
+            for (dep_name, dep) in &resolved_pkg.source_package.dependencies {
```

Comment:
The semantics of this check have no changed slightly -- previously it was ensuring that there was a consistency between nodes in the graph data structure and in the package table, now it's doing that consistency check by looking at the dependencies in the package table which is slightly different.  Could you go into the details of why this change was necessary? (It doesn't seem wrong to me, it's just a little less direct, so it would help me to understand).

Reply:
Ah that makes sense, thanks for clarifying. To say it back to make sure I've understood:

The previous implementation of the check confirmed that the keys in the package table matched the names in manifests, in order to check that the names given to dependencies matched the names packages gave themselves.  Now that the names in the package table are derived from the manifests, via the name resolution hook, the old implementation would be performing a redundant check so we have to literally check that the dependencies of all packages point back to a valid package in the table.

Reply:
>  the old implementation would be performing a redundant check so we have to literally check that the dependencies of all packages point back to a valid package in the table.

The old implementation is performing package name from table (which is loaded from dependency section of parent) to package name from manifest equality check. As it is this wouldn't work with resolution hook because package name from table (which is now a resolved name) is different from one in manifest. And if you do a package name from table (resolved name) to `custom_resolve_pkg_name(&self.source_package)` check this is indeed a redundand check.

So what we need is dependency name from depency section in parent manfiest to name in manifest check for sanity, otherwise someone could have a `DeepBook12313 = { ... }` for the `DeepBook` package and everything would work (since these names are now irrelevant for dependency resolution unless your custom hook relies on that).